src/ui/Thumbnail/Thumbnail.ts

```diff
@@ -70,7 +71,7 @@ export class Thumbnail extends Component {
    * automatically resolved (with a slower execution time).
    * @param result The result to associate the component with.
    */
-  constructor(public element: HTMLElement, public options?: IThumbnailOptions, public bindings?: IResultsComponentBindings, public result?: IQueryResult) {
+  constructor(public element: HTMLElement, public options?: IThumbnailOptions, public bindings?: IResultsComponentBindings, public result?: IQueryResult, private IconModule = Icon) {
```

Comment:
I am not a fan of doing that only for UT, only when it's *absolutely* necessary (for example, to inject a different modal box module which would make it impossible to test otherwise). The modal box is special in that it tries to open an iframe, which is annoying when we are executing lots of tests.

I feel that in this case, we could very well test with the real Icon module. It's not intrusive.

The test could become something like `expect(get(test.cmp.root.find('.CoveoIcon') instanceof Icon)).toBe(true)` or something like that.